### PR TITLE
Delete dev stage and legacy Chalice staging/prod deployment

### PR DESCRIPTION
**Pull request by mbarrien**
See other PR for more details. This PR must be merged first.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -86,7 +86,7 @@
     "e2e": "jest --config jest/e2e.config.js",
     "e2e-localProd": "TEST_ENV=localProd jest --config jest/e2e.config.js",
     "e2e-localProd-ci": "DEBUG=pw:api TEST_ENV=localProd jest --config jest/e2e.config.js",
-    "e2e-dev": "TEST_ENV=dev jest --config jest/e2e.config.js",
```
Comment (seve) [inline frontend/package.json:89]: There is no connected TEST_ENV in `frontend/tests/common/constants.ts` should we point the test env to a URL or delete the e2e option entirely?

Comment (mbarrien) [reply frontend/package.json:89]: Remove e2e-test

**Review by Bento007 — APPROVED** [review]

**Review by seve — COMMENTED** [review]
Can we also rename push-tests.yml and the workflow itself to better encompass what it does.

**Review by MDunitz — APPROVED** [review]
lgtm

**Pull request merged**